### PR TITLE
Added ESP8266 Support

### PR DESCRIPTION
**Pull request by ich423**
I have added support for the ESP8266.

I also added support for freeRTOS (only for ESP32).

### CODE REVIEW

File: 0TA_Template_Sketch/OTA.h [hunk 1]
```diff
@@ -1,12 +1,27 @@
...
+
+
 #include <WiFiUdp.h>
 #include <ArduinoOTA.h>
 #include <credentials.h>
 
 const char* ssid = mySSID;
 const char* password = myPASSWORD;
 
+#ifdef ESP32_RTOS && ESP32
```
Comment (Crazor) [inline 0TA_Template_Sketch/OTA.h:17]: This is not how you check for two defines. Notice the compiler warning: `warning: extra tokens at end of #ifdef directive`.
The correct way would be: `#if defined(ESP32_RTOS) && defined(ESP32)`.

File: 0TA_Template_Sketch/0TA_Template_Sketch.ino [hunk 2]
```diff
@@ -12,8 +15,12 @@ void setup() {
 }
 
 void loop() {
+#ifndef ESP32_RTOS && ESP32
```
Comment (Crazor) [inline 0TA_Template_Sketch/0TA_Template_Sketch.ino:18]: I'm not sure if this should be `#if !defined(ESP32_RTOS) && !defined(ESP32)` or `#if !defined(ESP32_RTOS) || !defined(ESP32)`. See my next comment for an explanation of the `defined()` syntax.

File: 0TA_Template_Sketch/OTA.h [hunk 3]
```diff
@@ -27,34 +42,44 @@ void setupOTA() {
...
+    else if (error == OTA_END_ERROR) Serial.println("End Failed");
+  });
 
   ArduinoOTA.begin();
 
   Serial.println("OTA Initialized");
   Serial.print("IP address: ");
   Serial.println(WiFi.localIP());
+
+#ifdef ESP32_RTOS && ESP32
```
Comment (Crazor) [inline 0TA_Template_Sketch/OTA.h:76]: Same as the previous comment.

File: 0TA_Template_Sketch/OTA.h (hunk 1 above)
Comment (ich423) [reply 0TA_Template_Sketch/OTA.h]: I have fixed that.

File: 0TA_Template_Sketch/0TA_Template_Sketch.ino (hunk 2 above)
Comment (ich423) [reply 0TA_Template_Sketch/0TA_Template_Sketch.ino]: I have fixed that.

File: 0TA_Template_Sketch/OTA.h (hunk 3 above)
Comment (ich423) [reply 0TA_Template_Sketch/OTA.h]: I have fixed that.

**Pull request merged**